client/components/jetpack/upsell-product-card/test/index.jsx

```diff
@@ -0,0 +1,228 @@
+					},
+					activePartnerKey: 1,
+				},
+				licenses: {
+					paginated: null,
+				},
+			},
+			sites: {
+				products: {
+					2916284: {
+						data: {
+							jetpack_scan: { available: true },
+						},
+					},
+				},
+			},
+			currentUser: {
+				user: {
+					has_jetpack_partner_access: true,
+				},
+			},
+		};
+		const queryClient = new QueryClient();
+		const mockStore = configureStore();
+		const store = mockStore( secondState );
+
+		const mockOnClick = jest.fn();
+		// Render component
+		render(
+			<Provider store={ store }>
+				<QueryClientProvider client={ queryClient }>
+					<UpsellProductCard
+						featureType={ FEATURE_TYPE_JETPACK_SCAN }
+						nonManageProductSlug={ PRODUCT_JETPACK_SCAN }
+						siteId={ 2916284 }
+						onCtaButtonClick={ mockOnClick }
+					/>
+				</QueryClientProvider>
+			</Provider>
+		);
+
+		expect( screen.getByRole( 'link' ) ).toHaveClass( 'jetpack-rna-action-card__button' );
+		expect( screen.getByRole( 'link' ) ).toHaveAttribute( 'href', '#' );
+
+		fireEvent.click( screen.getByText( 'Add Jetpack Scan' ) );
+
+		expect( mockOnClick ).toHaveBeenCalledTimes( 1 );
+	} );
+
+	test( 'lightbox buttons and close icon act as expected', () => {
+		const createFakeApiProduct = {
+			name: 'Jetpack Scan Daily',
+			slug: 'jetpack-scan',
+			product_id: 2106,
+			currency: 'USD',
+			amount: 0,
+			price_interval: 'month',
+			family_slug: 'jetpack-scan',
+		};
+
+		const secondState = {
+			partnerPortal: {
+				partner: {
+					current: {
+						keys: { id: 1 },
+					},
+					activePartnerKey: 1,
+				},
+				licenses: {
+					paginated: null,
+				},
+			},
+			sites: {
+				products: {
+					2916284: {
+						data: {
+							jetpack_scan: { available: true },
+						},
+					},
+				},
+			},
+			currentUser: {
+				user: {
+					has_jetpack_partner_access: true,
+				},
+			},
+			productsList: {
+				items: {
+					'jetpack-scan': createFakeApiProduct,
+				},
+			},
+		};
+
+		const queryClient = new QueryClient();
+		const mockStore = configureStore();
+		const store = mockStore( secondState );
+
+		const mockOnClose = jest.fn();
+
+		// Render component
+		render(
+			<Provider store={ store }>
+				<QueryClientProvider client={ queryClient }>
+					<SingleSiteUpsellLightbox
+						manageProduct={ createFakeApiProduct }
+						onClose={ mockOnClose }
+						nonManageProductSlug="jetpack_scan"
+						nonManageProductPrice={ null }
+						partnerCanIssueLicense={ true }
+						siteId={ 2916284 }
+					/>
+				</QueryClientProvider>
+			</Provider>
+		);
+
+		const lightboxElement = document.querySelectorAll( '.jetpack-lightbox__content-wrapper' );
+		const issueLicenseButton = within( lightboxElement[ 0 ] ).getByText( 'Issue license' );
+		const closeIcon = screen.getByLabelText( 'Close' );
+
+		expect( screen.getByText( 'Purchase via Jetpack.com' ) ).toHaveAttribute(
+			'href',
+			'https://wordpress.com/checkout/jetpack/jetpack_scan?redirect_to=https%3A%2F%2Fexample.com%2F'
```

Comment:
Not sure if this matters, but the URL seems incorrect here; on my live site I have something like this:

https://wordpress.com/checkout/objective-recluse.jurassic.ninja/jetpack_scan?redirect_to=http%3A%2F%2Fjetpack.cloud.localhost%3A3000%2Fscan%2Fexample.com%23&source=jetpack-plans&checkoutBackUrl=http%3A%2F%2Fjetpack.cloud.localhost%3A3000%2Fscan%2Fexample.com%23

Also, it seems `2916284` is for `example.wordpress.com`, not `example.com`. Again, not sure if that matters.

Reply:
For future reference, it seems Jest is configured to use `example.com` as the default URL, so unit tests would have that in as the default `window.location`:
https://github.com/Automattic/wp-calypso/blob/3449c4e81612c5def876aac2398f9936b4599901/test/client/jest.config.js#L17

The site ID has nothing to do with it.

The other params are likely missing in tests due to environment differences.